### PR TITLE
Transparent Encryption with IPSec Getting Started Guide

### PR DESCRIPTION
**Pull request by jrfastab**
Initial Transparent Encryption via IPSec getting started advance topics doc.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7034)

### CODE REVIEW

File: Documentation/gettingstarted/encryption.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+****************************
+Using Transparent Encryption
+****************************
+
+This guide explains how to configure Cilium to use IPSec based transparent
+encryption using etcd secrets to distribute the IPSec keys. After this
+configuration is complete all traffic between Cilium
+managed endpoints, as well as Cilium managed host traffic, will be encrypted
+using IPSec. This guide uses etcd secrets to distribute keys alternatively
```
Comment (tgraf) [inline Documentation/gettingstarted/encryption.rst]: s/etcd/Kubernetes/

Comment (tgraf) [reply Documentation/gettingstarted/encryption.rst]: keys. Alternatively,

File: Documentation/gettingstarted/encryption.rst [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.9/cilium-ds.yaml
+
+  .. group-tab:: K8s 1.8
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.8/cilium-ds.yaml
+
+First create a yaml file for the IPSec keys to be stored as a etcd secret.
```
Comment (tgraf) [inline Documentation/gettingstarted/encryption.rst]: s/etcd/Kubernetes/

File: Documentation/gettingstarted/encryption.rst [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+.. parsed-literal::
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    name: cilium-ipsec-keys
+  type: Opaque
+  stringData:
+
+Next we will generate the necessary IPSec keys which will be distributed as an
+etcd secret using the 'cilium-ipsec-keys.yaml' file. In this example we use
```
Comment (tgraf) [inline Documentation/gettingstarted/encryption.rst]: s/etcd/Kubernetes/ ;-)

File: Documentation/gettingstarted/encryption.rst [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+AES-CBC with HMAC-256 (hash based authentication code), but any of the supported
+Linux algorithms may be used. To generate use the following
+
+.. parsed-literal::
+  KEY1=0x`dd if=/dev/urandom count=32 bs=1 2> /dev/null| xxd -p -c 64`
```
Comment (tgraf) [inline Documentation/gettingstarted/encryption.rst:81]: There is also:
```
kubectl create --dry-run=true secret generic cilium-ipsec-keys --from-file=... -o yaml > secret.yaml
```

but this is fine as well

File: Documentation/gettingstarted/encryption.rst [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  KEY2=0x`dd if=/dev/urandom count=32 bs=1 2> /dev/null| xxd -p -c 64`
+  echo "keys: \\"hmac(sha256) $KEY1 cbc(aes) $KEY2\\"" >> cilium-ipsec-keys.yaml
+
+From the 'cilium-ipsec-keys.yaml' file generate an etcd secret.
```
Comment (tgraf) [inline Documentation/gettingstarted/encryption.rst]: s/...

File: Documentation/gettingstarted/encryption.rst [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    template:
+      spec:
+        containers:
+        - name: cilium-agent
+          args:
+          - "--debug=$(CILIUM_DEBUG)"
+          - "--kvstore=etcd"
+          - "--kvstore-opt=etcd.config=/var/lib/etcd-config/etcd.config"
+          - "--enable-ipsec"
+          - "--ipsec-key-file=/etc/ipsec/keys"
```
Comment (tgraf) [inline Documentation/gettingstarted/encryption.rst:115]: This is OK for now but we should use an environment variable instead going forward

**Review by tgraf — CHANGES_REQUESTED** [review]
Looks great and simple. Just needs a global s/etcd secrets/Kubernetes secrets/

File: Documentation/gettingstarted/encryption.rst (hunk 6 above)
Comment (jrfastab) [reply Documentation/gettingstarted/encryption.rst:115]: OK will add it to the list.

**Review by tgraf — APPROVED** [review]

**Pull request merged**